app/hooks/useChallenge.ts

```diff
@@ -0,0 +1,24 @@
+import { useQuery } from "@tanstack/react-query";
```

Comment:
Don't needs hooks for reads on data that will be indexed in prisma, we can just use remix loaders for that.

Reply:
Sebastian and I were thinking the same thing